src/pydata_sphinx_theme/toctree.py

```diff
@@ -373,40 +369,87 @@ def add_collapse_checkboxes(soup: BeautifulSoup) -> None:
         # Add a class to indicate that this has children.
         element["class"] = classes + ["has-children"]
 
-        # We're gonna add a checkbox.
-        toctree_checkbox_count += 1
-        checkbox_name = f"toctree-checkbox-{toctree_checkbox_count}"
-
-        # Add the "label" for the checkbox which will get filled.
         if soup.new_tag is None:
             continue
 
-        label = soup.new_tag(
-            "label", attrs={"for": checkbox_name, "class": "toctree-toggle"}
-        )
-        label.append(soup.new_tag("i", attrs={"class": "fa-solid fa-chevron-down"}))
-        if "toctree-l0" in classes:
-            # making label cover the whole caption text with css
-            label["class"] = "label-parts"
-        element.insert(1, label)
-
-        # Add the checkbox that's used to store expanded/collapsed state.
-        checkbox = soup.new_tag(
-            "input",
+        # For table of contents nodes that have subtrees, we modify the HTML so
+        # that the subtree can be expanded or collapsed in the browser.
+        #
+        # The HTML markup tree at the parent node starts with this structure:
+        #
+        # li.has-children
+        #   a.reference or p.caption
+        #   ul
+        #
+        # Note the first child is p.caption only if this node is a section heading and
+        # the theme option show_nav_level is set to 0.
+        #
+        # Now we modify the tree structure in one of two ways.
+        #
+        # (1) If the node holds a section heading AND show_nav_level == 0, the HTML
+        # tree will be modified like so:
+        #
+        # li.has-children
+        #   details
+        #       summary
+        #           p.caption
+        #           .toctree-toggle
+        #       ul
+        #
+        # (2) Otherwise, if the node holds a link to a page in the docs:
```

Comment:
I am struggling to grapple some of the specifics so apologies if this does not make sense at all. Why does it only matter for case 1) that `show_nav_level == 0` and not here?

Reply:
Good question. This is fairly complicated to explain.

Let's begin by reviewing the levels of navigation on the site:

- sections within site (user guide, contributor guide, examples)
- parts within a section (within contributor guide: contributor guide, team practices, about the project)
- pages within a part (e.g., contributor guide / contributor guide / get started with development)
- (optional) sub-pages, and sub-pages of sub-pages, and so on
- heading within page or sub-page (e.g. contributor guide / contributor guide / get started with development / workflow for contributing changes)

The first level is put in the top nav, all the other levels are put in the left sidebar, except the last level, the within (sub)page level, which is put in the right sidebar.

Normally, when the table of contents is generated for the left sidebar, several nested tree structures using HTML lists of nested lists are created to capture each group of pages and sub-pages within a part, but the parts themselves are rendered outside of these nested list structures.

However, when `show_nav_level == 0` the `generate_toctree_html()` function modifies the HTML so that the section parts become part of the nested list-of-lists structure.

Note that this part of the code you've asked about is in a function called `add_collapse_checkboxes`. This function goes through every list item in the nested list structure it receives, and if that list item contains a list, then it decorates the HTML so that that list item can be toggled open or closed.

But here's the catch. Unlike the pages and sub-pages (nav level > 0), the section parts (nav level = 0) are not links; there is no URL and no page in the site that maps to any section part.

This is the key issue. Every node in the sidebar table of contents is a link **except** for the section headings, which are all 0th-level nodes in the sidebar (or 1st-level nodes if you consider the "Section Navigation" heading at the top to be the root node for the entire sidebar TOC).

To reiterate, the key difference is that section nodes are not links, whereas page and sub-page nodes are. When a node is a link, we have to think about whether we want to put the link inside the summary tag or not. Why? If we put a link in a summary tag, we introduce an ambiguity: is the link click to expand/collapse or to navigate to a new page? I made the decision that the link click should be a link click and not an expand/collapse. However, I saw no reason why the user shouldn't be able to click the text of the section part to expand/collapse the details/summary, since a section part is not a link.

When `show_nav_level > 0`, the section parts are rendered outside of the list-of-lists structure, so they are never even visited by this `add_collapse_checkboxes` function.

Reply:
I just updated the code comment here. Hopefully, it's slightly clearer, or at least more precise, now.

Reply:
Thanks for the in-depth explanation; this makes sense. I went through the rest of the PR again and could not spot any obvious gotchas. So, I am happy to approve it as is.

That being said I know @drammock is a lot more familiar with the whole parsing/HTML generation side of things so I want to give him the chance to go over this or I can merge in a couple of days.